File: modules/web/src/com/company/demo/web/extension/IframeOptions.java
package com.company.demo.web.extension;

import com.vaadin.annotations.JavaScript;
import com.vaadin.server.AbstractJavaScriptExtension;
import com.vaadin.ui.BrowserFrame;

@JavaScript("iframe-options.js")
public class IframeOptions extends AbstractJavaScriptExtension {

    public IframeOptions(BrowserFrame target) {
        super(target);
    }

    public void setSandbox(String sandbox) {
        getState().sandbox = sandbox;
    }

    @Override
    protected IframeOptionsState getState() {
        return (IframeOptionsState) super.getState();
    }

    @Override
    protected IframeOptionsState getState(boolean markAsDirty) {
        return (IframeOptionsState) super.getState(markAsDirty);
    }
}

File: modules/web/src/com/company/demo/web/extension/iframe-options.js
com_company_demo_web_extension_IframeOptions = function () {
    var self = this;

    this.onStateChange = function () {
        console.log("Sandbox onStateChange");
        var iframe = self.getElement(self.getParentId()).firstElementChild;

        if (this.getState().sandbox) {
            console.log("Sandbox is set");
            console.log(iframe);

            // set to iframe
            iframe.setAttribute("sandbox", this.getState().sandbox);
        }
    }
};